### PR TITLE
feat(protocol): L2/L3 contracts proxied

### LINKED ISSUE #13722: Update L2 genesis generation script to deploy L2 contracts behind proxies

### PR DESCRIPTION
**Pull request by cyberhorsey**

### CODE REVIEW

File: packages/protocol/test/genesis/test_config.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "contractOwner": "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
+  "contractAdmin": "0xc0ffee254729296a45a3885639AC7E10F9d54979",
```
Comment (cyberhorsey) [inline packages/protocol/test/genesis/test_config.json:3]: this is a random address, not sure if it matters here

File: packages/protocol/utils/generate_genesis/taikoL2.ts [hunk 2]
```diff
@@ -59,15 +66,22 @@ export async function deployTaikoL2(
 
         // pre-mint ETHs for EtherVault contract
         alloc[contractConfig.address].balance =
-            contractName === "EtherVault"
+            contractName === "ProxiedEtherVault"
```
Comment (davidtaikocha) [inline packages/protocol/utils/generate_genesis/taikoL2.ts:69]: Seems we need to pre-mint L2 ETHs to the `EtherVaultProxy`?

File: packages/protocol/test/genesis/test_config.json (hunk 1 above)
Comment (davidtaikocha) [reply packages/protocol/test/genesis/test_config.json:3]: this json is only for testing, so i think its ok.

File: packages/protocol/contracts/thirdparty/TransparentUpgradeableProxy.sol [hunk 3]
```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MIT
```
Comment (dantaik) [inline packages/protocol/contracts/thirdparty/TransparentUpgradeableProxy.sol:1]: Jeff, why don't we use something from openzapllin's library?

File: packages/protocol/utils/generate_genesis/taikoL2.ts [hunk 4]
```diff
@@ -9,12 +9,18 @@ const {
 } = require("@defi-wonderland/smock/dist/src/utils");
 const ARTIFACTS_PATH = path.join(__dirname, "../../out");
 
+const IMPLEMENTATION_SLOT =
+    "0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc";
```
Comment (dantaik) [inline packages/protocol/utils/generate_genesis/taikoL2.ts:13]: How this is calculated?

File: packages/protocol/contracts/thirdparty/TransparentUpgradeableProxy.sol (hunk 3 above)
Comment (cyberhorsey) [reply packages/protocol/contracts/thirdparty/TransparentUpgradeableProxy.sol:1]: this is from openzeppelin, its a copy

File: packages/protocol/utils/generate_genesis/taikoL2.ts (hunk 2 above)
Comment (cyberhorsey) [reply packages/protocol/utils/generate_genesis/taikoL2.ts:69]: done!

**Review by dantaik — APPROVED** [review]

**Review by davidtaikocha — APPROVED** [review]

**Pull request merged**